inbm/dispatcher-agent/dispatcher/sota/sota.py

```diff
@@ -356,7 +357,7 @@ def execute_from_manifest(self,
                     self._update_logger.status = OTA_PENDING
                     self._update_logger.error = ""
                 self._update_logger.save_log()
-                if self.sota_mode == 'download-only' or self._device_reboot in ["No", "N", "n", "no", "NO"]:  # pragma: no cover
+                if self.sota_mode == 'download-only' or rebooter.is_reboot(self._device_reboot):
                     self._dispatcher_broker.telemetry("No reboot (SOTA pass)")
                 else:
                     self._dispatcher_broker.telemetry("Going to reboot (SOTA pass)")
```

Comment:
**Suggestion:** Use a more robust check for reboot conditions that does not rely on string comparison, which can be error-prone and less maintainable. [enhancement]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
                if self.sota_mode == 'download-only' or not rebooter.is_reboot_required():
                    self._dispatcher_broker.telemetry("No reboot (SOTA pass)")
                else:
                    self._dispatcher_broker.telemetry("Going to reboot (SOTA pass)")
```